bob-apps/bin/blob_recovery_tool/structs.rs

```diff
@@ -81,6 +81,13 @@ impl Validatable for BlobHeader {
     }
 }
 
+impl BlobHeader {
+    pub(crate) fn migrate(mut self, version: u32) -> AnyResult<Self> {
```

Comment:
The signature should be the same as in `Record.migrate`. It should receive source_version and target_version as parameters. Also the behavior should be the same: you need to explicitly restrict possible migrations